app/models/preserved_object_primary_moab.rb

```diff
@@ -0,0 +1,7 @@
+class PreservedObjectPrimaryMoab < ApplicationRecord
+  belongs_to :preserved_object, inverse_of: :preserved_objects_primary_moabs
+  belongs_to :complete_moab, inverse_of: :preserved_objects_primary_moabs
+
+  validates :preserved_object, presence: { strict: true }, uniqueness: { strict: true }
```

Comment:
I might be in favor of leaving the uniqueness restriction in the database layer rather than in the application. We've made similar moves (towards DB constraints and away from model validations related to uniqueness) in a couple of other codebases recently to help with race conditions, and to avoid having to rescue/deal with both `ActiveRecord::RecordInvalid` and `ActiveRecord::RecordNotUnique`.

Reply:
Agreed, this is belt and suspenders situation.

Reply:
For that matter, I'm not sure the presence validations are useful here either for the same reason.

Reply:
removed.  @jmartin-sul and I discussed that there might be pushback on this point.  John prefers the error messages from Rails, I don't care myself.   I'm actually wondering if this model is going to be useful, but trust that John has a use for it in the future (and he did ask for it).

Reply:
I also discussed w/ @jcoyne and he pointed out the value of the model is the ability to access the created_at/updated_at values in the join table.

Reply:
avoiding the race condition is a compelling argument, hadn't considered that.  tbh i was mostly in the habit of using both because validations seemed like the idiomatic/rails-y way to do things, but the DB constraint always felt like the must-have.  definitely happy to just use DB constraints going forward.

would also be totally fine to file a tech debt ticket (prob backlog?) to remove the duplicative rails uniqueness and presence validations since this project has been pretty good about DB constraints overall afaict, other than a few multi-column uniqueness constraints we've missed the opportunity to add.